src/components/victory-bar.jsx

```diff
@@ -265,24 +276,28 @@ class VBar extends React.Component {
   }
 
   _getAttributes(props, index) {
-    const attributes = props.dataAttributes && props.dataAttributes[index] ?
+    let attributes = props.dataAttributes && props.dataAttributes[index] ?
       props.dataAttributes[index] : props.dataAttributes;
+    if (attributes) {
+      attributes.fill = attributes.fill || this.getColor(index);
+    } else {
+      attributes = {fill: this.getColor(index)};
+    }
+
     const requiredAttributes = {
       name: attributes && attributes.name ? attributes.name : "data-" + index
     };
     return _.merge(requiredAttributes, attributes);
   }
 
-  containsStrings(collection) {
-    return _.some(collection, function (item) {
-      return _.isString(item);
-    });
+  getColor(index) {
+    return Util.style.getColorScale(this.props.colorScale)[index % 5];
```

Comment:
Please remove the hardcoded 5 here. It should work for color scales of whatever length.